### ISSUE TITLE
genius-1.0.21-2: Undefined symbols for architecture x86_64

### ISSUE THREAD

**Issue by cooljeanius** [fink/fink-distributions#1081]
Looks like that issue with using `libtool` to link on Big Sur again:
```
libtool: link: gcc  -o .libs/libtestplugin.so -bundle  .libs/testplugin.o   -L/opt/sw/lib/0 -L/opt/sw/lib /opt/sw/lib/libmpfr.dylib  -O2 -Wl,-headerpad_max_install_names  
ld: warning: directory not found for option '-L/opt/sw/lib/0'
Undefined symbols for architecture x86_64:
  "_d_add_named_args", referenced from:
      _open in testplugin.o
  "_d_addfunc", referenced from:
      _open in testplugin.o
  "_d_intern", referenced from:
      _open in testplugin.o
  "_d_makebifunc", referenced from:
      _open in testplugin.o
  "_gel_add_category", referenced from:
      _open in testplugin.o
  "_gel_add_description", referenced from:
      _open in testplugin.o
  "_gel_evalexp", referenced from:
      _open in testplugin.o
  "_gel_main_out", referenced from:
      _open in testplugin.o
      _TestPluginFunction_op in testplugin.o
  "_gel_makenum_null", referenced from:
      _TestPluginFunction_op in testplugin.o
  "_gel_output_printf", referenced from:
      _open in testplugin.o
      _TestPluginFunction_op in testplugin.o
  "_libintl_gettext", referenced from:
      _open in testplugin.o
      _TestPluginFunction_op in testplugin.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[3]: *** [Makefile:736: libtestplugin.la] Error 1
make[3]: *** Waiting for unfinished jobs....
mv -f .deps/funclib.Tpo .deps/funclib.Po
libtool: link: gcc -finline-functions -frename-registers -g -O2 -Wl,-headerpad_max_install_names -o genius-readline-helper-fifo genius_lists.o genius-readline-helper.o  -L/opt/sw/lib/0 -L/opt/sw/lib ../ve/libvicious.a -lm /opt/sw/lib/libgmodule-2.0.dylib /opt/sw/lib/libglib-2.0.dylib /opt/sw/lib/libintl.dylib -lreadline -ltermcap /opt/sw/lib/libmpfr.dylib
ld: warning: directory not found for option '-L/opt/sw/lib/0'
1 warning generated.
mv -f .deps/mpwrap.Tpo .deps/mpwrap.Po
mv -f .deps/eval.Tpo .deps/eval.Po
2 warnings generated.
mv -f .deps/graphing.Tpo .deps/graphing.Po
make[3]: Leaving directory '/opt/sw/src/fink.build/genius-1.0.21-2/genius-1.0.21/src'
make[2]: *** [Makefile:665: all] Error 2
make[2]: Leaving directory '/opt/sw/src/fink.build/genius-1.0.21-2/genius-1.0.21/src'
make[1]: *** [Makefile:577: all-recursive] Error 1
make[1]: Leaving directory '/opt/sw/src/fink.build/genius-1.0.21-2/genius-1.0.21'
make: *** [Makefile:443: all] Error 2
### execution of make failed, exit code 2
### execution of /tmp/fink.N7I99 failed, exit code 2
Removing runtime build-lock...
Removing build-lock package...
/opt/sw/bin/dpkg-lockwait -r fink-buildlock-genius-1.0.21-2
```
cc @dmacks

**Comment by dmacks** [comment]
`-L/opt/sw/lib/0` is a harmless bug in fink itself that thesin and I fixed over the past few days. The symbol error is probably due to the tarball having an ancient libtool that doesn't know about modern OS. Various existing packages have solved this with different specific patches depending on the libtool vintage. I'm on an older OS so I can't test as directly.

**Comment by cooljeanius** [comment]
> The symbol error is probably due to the tarball having an ancient libtool that doesn't know about modern OS. Various existing packages have solved this with different specific patches depending on the libtool vintage.

Yeah see for example @nieder's commit 0d04f6b to fix #1069

**Comment by dmacks** [comment]
[e80c5d6](https://github.com/fink/fink-distributions/commit/e80c5d68d585fcc57c7c470415ce83a0bab5005c)
